sympy/solvers/ode.py

```diff
@@ -4049,8 +4049,8 @@ def ode_order_reducing_substitution(eq, func, order, match):
         if not eq.has(g(x)):
             break
     eq = eq.subs(f(x).diff(x, match['var']), g(x))
-    eq = dsolve(eq)
-    eq = dsolve(eq.subs(g(x), f(x).diff(x, match['var'])))
+    eq = dsolve(eq, g(x))
+    eq = dsolve(eq.subs(g(x), f(x).diff(x, match['var'])), f(x))
```

Comment:
I wonder if this second `dsolve` should really be `Integral` rather than `dsolve`. The advantage I guess of using `dsolve`  is that it can work if `dsolve(eq, g(x))` returns an implicit equation for `g(x)`. If the code is intended to work for that case then there should be a test for it to see that it does. If the code is only intended to work for explicit solutions then we should check that `eq` is in explicit form before this second call to `dsolve`.

Reply:
Well, I don't think it will matter either way.But Integral would require a loop e.g, f'''(x) = sin x requires 3 integration.So let dsolve do the task.

Reply:
A loop using `Integral` (not `integrate`) is much more efficient than calling `dsolve`.

Reply:
Oh sorry I didn't read it full and in my brain still previous code existed and in reference to that I wrote that.

Reply:
Hey, are you acquainted to this
https://en.wikipedia.org/wiki/Cauchy_formula_for_repeated_integration.
This can be handy in solving many integrals(atleast it can have wonderful applications in this function) and if you see the a that is there ,if you club all terms it results in a constant C.Should I make such a function?

Reply:
And doesn't Integral only uses integral signs but does not integrate?

Reply:
Yes, `Integral(x,x)` is unchanged but `integrate(x,x) = x**2/2`.